### PR TITLE
[Compute] Fix #5258: `az image copy`: Fix `expected not NoneType` error

### LINKED ISSUE #5258: az image copy failed with error
## Describe the bug

**Command Name**
`az image copy
Extension Name: image-copy-extension. Version: 0.2.9.`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
expected str, bytes or os.PathLike object, not NoneType
Traceback (most recent call last):
  File "/opt/az/lib/python3.10/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 663, in execute
    raise ex
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 697, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/__init__.py", line 333, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.10/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/root/.azure/cliextensions/image-copy-extension/azext_imagecopy/custom.py", line 108, in imagecopy
    run_cli_command(cli_cmd)
  File "/root/.azure/cliextensions/image-copy-extension/azext_imagecopy/cli_utils.py", line 21, in run_cli_command
    cmd_output = check_output(cmd, stderr=STDOUT, universal_newlines=True)
  File "/opt/az/lib/python3.10/subprocess.py", line 420, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/opt/az/lib/python3.10/subprocess.py", line 501, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/opt/az/lib/python3.10/subprocess.py", line 969, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/az/lib/python3.10/subprocess.py", line 1778, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
TypeError: expected str, bytes or os.PathLike object, not NoneType

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az image copy --source-resource-group {} --source-object-name {} --target-location {} --target-resource-group {} --source-type {}`

## Environment Summary
```
Linux-5.13.0-30-generic-x86_64-with-glibc2.31, Ubuntu 20.04.4 LTS
Python 3.10.5
Installer: DEB

azure-cli 2.39.0

Extensions:
image-copy-extension 0.2.9

Dependencies:
msal 1.18.0b1
azure-mgmt-resource 21.1.0b1
```

### PR DESCRIPTION
**Pull request by yanzhudd**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Description

The logic of `az image copy` command is consisted by several Azure CLI commands. The `az image show` command is called first, and then the fields in the output will be used to call other commands later. 
This error mainly due to the `hyperVGeneration` field of `az image show` command output is null value and then passed to call other commands.
Issue - close: #5258

### Related command

`az image copy`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/image-copy/azext_imagecopy/create_target.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 def create_target_image(cmd, location, transient_resource_group_name, source_type, source_object_name,
                         source_os_disk_snapshot_name, source_os_disk_snapshot_url, source_os_type,
                         target_resource_group_name, azure_pool_frequency, tags, target_name, target_subscription,
-                        export_as_snapshot, timeout, hyper_v_generation='V1'):
+                        export_as_snapshot, timeout, hyper_v_generation=None):
```
Comment (zhoxing-ms) [inline src/image-copy/azext_imagecopy/create_target.py:24]: May I ask why do we need to modify the default value? Please note that modifying the default value of parameters may cause breaking change to users

Comment (yanzhudd) [reply src/image-copy/azext_imagecopy/create_target.py:24]: > May I ask why do we need to modify the default value? Please note that modifying the default value of parameters may cause breaking change to users

okay, I will restore the default value to `V1`

**Review by zhoxing-ms — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by wangzelin007 — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**